app/views/compute_resources_vms/form/libvirt/_base.html.erb

```diff
@@ -24,3 +24,27 @@
 </div>
 
 <%= compute_specific_js(compute_resource, "nic_info") %>
+
+<%= select_f f, :os_firmware,
+                Foreman::Model::Libvirt.firmware_types,
+                :first,
+                :last,
+                {},
+                { :label => _("Firmware"),
+                  :label_size => "col-md-2",
+                  :onchange => "tfm.computeResource.libvirt.firmwareSelected(this);",
+                }
+%>
+<%
+  feature_attrs = ActiveSupport::HashWithIndifferentAccess.new(f.object.os_firmware_features)
+  is_bios = f.object.os_firmware == 'bios'
+%>
+
+<div id="os_firmware_features">
+<%= f.fields_for :os_firmware_features do |ff|
+    secure_field = checkbox_f(ff, 'secure-boot', { checked: feature_attrs['secure-boot'] == 'yes', label: _("Secure Boot"), disabled: is_bios }, 'yes', 'no')
+    keys_field = checkbox_f(ff, 'enrolled-keys', { checked: feature_attrs['enrolled-keys'] == 'yes', label: _("Enrolled keys"), disabled: is_bios }, 'yes', 'no')
+    secure_field + keys_field
+  end
+%>
+</div>
```

Comment:
I think we should hide these options when editing a host and display them only when creating a new host. Since there is no way to edit these fields and we don’t know the actual values (we’re displaying the default ones), I don’t see a reason to include them in the edit form.